packages/contracts-bedrock/scripts/slither.sh

```diff
@@ -74,7 +73,7 @@ fi
 
 # If slither failed to generate a report, exit with an error.
 if [ ! -f "$SLITHER_REPORT" ]; then
-  echo "Slither output:\n$SLITHER_OUTPUT"
+  printf "Slither output:\n%s\n" "$SLITHER_OUTPUT"
```

Comment:
This definitely seems worse to me.

Reply:
I'm an `echo` guy myself.

Reply:
I also typically prefer `echo`, but @ajsutton tested (see below [comments](https://github.com/ethereum-optimism/optimism/pull/8594#issuecomment-1853044450)) and `echo` does behave differently by platform, so we should stick with the `printf` approach

Reply:
Specifically it performs differently in zsh (I just keep forgetting Mac uses zsh by default...) - it appears to just never do what this code wants it to in bash.  The echo version of this would print:

```
Slither output:\n<output>
```

and the `\n` would not be a newline.  Of course you could just write it as:

```
echo "Slither output:"
echo "$SLITHER_OUTPUT"
```
which now that I think about it is probably the best option yet.

Reply:
> Of course you could just write it as:
> 
> ```
> echo "Slither output:"
> echo "$SLITHER_OUTPUT"
> ```
> which now that I think about it is probably the best option yet.

Oh of course, great point. I agree this is my favorite: more readable and we can keep `echo` everywhere instead of mixing the two

Reply:
Some of the changes here were from `echo -n` because in POSIX sh, echo flags are undefined per [SC3037](https://www.shellcheck.net/wiki/SC3037). For these, I think we'd still have to use printf

Reply:
I think the two that are remaining are in `entrypoint-l1.sh` which we can move to bash and thus take advantage of `echo -n`.  Must admit, I've never come across an echo that didn't support -n.  It works even on dash which provides a very minimal sh on ubuntu and doesn't support bash features.

Reply:
Resolving this thread since we now use bash and kept `echo -n`